src/bin/main.rs

```diff
@@ -305,20 +332,35 @@ fn main() {
         };
     }
 
+    let mut track_copy_dest = HashMap::new();
     if let Some(code_dir) = code_dir {
         if let Some(code_files) = code_files {
-            for code_file in &code_files {
-                eprintln!("Copying code file {:?}", code_file);
+            for (code_file, code_out_path) in &code_files {
+                if track_copy_dest.contains_key(code_out_path) {
```

Comment:
Using `match track_copy_dest.entry(code_out_path)` could probably remove one of the hash map look-ups.